### ISSUE TITLE
Copying .ts files and renaming them confuses TypeScript

### ISSUE THREAD

**Issue by oliversalzburg** [microsoft/vscode#105133]
Issue Type: <b>Bug</b>

I encounter this in other circumstances, but the easiest way to reproduce this for me is in this small repro project I was building recently: https://github.com/oliversalzburg/mixin-mikro-orm

I would usually run a `tsc --watch` as a task in VSC, but in this project a `yarn build:watch` would achieve the same thing.

When I create copies of `.ts` files, like those in the `entities` folder, and then rename them (to create a similar entity), this will sooner or later trip up TypeScript in VSC. This manifests in an error in the problems panel:

> File '/tmp/mixin-mikro-orm/src/entities/Other copy 2.ts' not found.

The error points to `tsconfig.json`. The file does not exist and I need to reload the TS server to recover from this.

VS Code version: Code 1.48.1 (3dd905126b34dcd4de81fa624eb3a8cbe7485f13, 2020-08-19T17:12:13.244Z)
OS version: Windows_NT x64 10.0.19041
Remote OS version: Linux x64 4.19.104-microsoft-standard
Remote OS version: Linux x64 4.19.104-microsoft-standard

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-6850K CPU @ 3.60GHz (12 x 3592)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|63.87GB (34.26GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|WSL: Ubuntu|
|OS|Linux x64 4.19.104-microsoft-standard|
|CPUs|Intel(R) Core(TM) i7-6850K CPU @ 3.60GHz (12 x 3591)|
|Memory (System)|50.11GB (35.04GB free)|
|VM|0%|

|Item|Value|
|---|---|
|Remote|WSL: Ubuntu|
|OS|Linux x64 4.19.104-microsoft-standard|
|CPUs|Intel(R) Core(TM) i7-6850K CPU @ 3.60GHz (12 x 3591)|
|Memory (System)|50.11GB (35.04GB free)|
|VM|0%|
</details><details><summary>Extensions (35)</summary>

Extension|Author (truncated)|Version
---|---|---
eslint-disable-snippets|drK|1.2.0
output-colorizer|IBM|0.1.2
vscode-docker|ms-|1.5.0
remote-containers|ms-|0.134.1
remote-ssh|ms-|0.51.0
remote-ssh-edit|ms-|0.51.0
remote-wsl|ms-|0.44.4
vscode-remote-extensionpack|ms-|0.20.0
vscode-icons|vsc|10.2.0
vscode-zipfs|arc|2.0.0
better-toml|bun|0.3.2
vscode-eslint|dba|2.1.8
xml|Dot|2.5.1
eslint-disable-snippets|drK|1.2.0
gitlens|eam|10.2.2
EditorConfig|Edi|0.15.1
prettier-vscode|esb|5.1.3
generateuuid|Fir|0.0.5
shell-format|fox|7.0.1
terraform|has|2.2.0
vscode-drawio|hed|0.7.2
vscode-esformatter|how|1.6.0
output-colorizer|IBM|0.1.2
code-beautifier|mic|2.3.3
vscode-lebab|mrm|2.6.0
vscode-puglint|mrm|2.3.0
vscode-docker|ms-|1.5.0
csharp|ms-|1.23.1
cpptools|ms-|0.29.0
hexeditor|ms-|1.2.1
js-debug-nightly|ms-|2020.8.1917
vscode-paste-image|mus|1.0.4
vscode-graphql|Pri|0.2.35
vscode-yaml|red|0.10.0
lit-plugin|run|1.2.1

(2 theme extensions excluded)

</details>

**Comment by oliversalzburg** [comment]
Moving files via Drag&Drop in the explorer also potentially causes the identical issue.

**Comment by mjbvz** [comment]
I can't reproduce this issue in the project. Can you please try collecting the TS Server log to help us investigate:

1. In the VS Code setting, set `"typescript.tsserver.log": "verbose",
1. Restart VS Code and reproduce the problem
1. With a TS file open, run the `TypeScript: Open TS Server log` command
1. This should open a large log file called `tsserver.log` 

If you can share the log it would help us understand what may be going on here.

 > **⚠️Warning:** The TypeScript log may include information from your workspace, including file paths and source code. If you have any concerns about posting this publicly on Github, just let me know and we can arrange something else. On our side, we only use these logs to investigate issues like this

**Comment by mjbvz** [comment]
Also be sure to test  using the [latest VS Code insiders build](https://code.visualstudio.com/insiders/) with all [extensions disabled](https://github.com/microsoft/vscode/wiki/Performance-Issues#run-with-extensions-disabled) just to make sure one of them is not causing this

**Comment by oliversalzburg** [comment]
I can't disable my extensions, because then I lose WSL support and that is where I last consistently reproduced this issue.

Here's a log with the issue reproduced. The error that was logged was:

```json
{
	"resource": "/home/oliver/fairmanager/backend/core-web3/tsconfig.json",
	"owner": "typescript",
	"severity": 8,
	"message": "File '/home/oliver/fairmanager/backend/core-web3/src/entities/sadg.ts' not found.",
	"source": "ts",
	"startLineNumber": 1,
	"startColumn": 1,
	"endLineNumber": 1,
	"endColumn": 2
}
```

[tsserver.log](https://github.com/microsoft/vscode/files/5148874/tsserver.log)

**Comment by mjbvz** [comment]
Thanks for the logs. I believe you are using the typescript version from yarn (`.yarn/sdks/typescript/lib/tsserver.js`). I believe they do some patching to make it support yarn2 and this has cause a few issues in the past. Can you check to see if this happens with a standard TypeScript version too?

**Comment by oliversalzburg** [comment]
That's correct. I can repro this with the builtin 4.0.2 without problems too though. Took like 6 renames now to trap it.

**Comment by mjbvz** [comment]
Are you using TypeScript 4.0.2 from yarn or the standard one?

**Comment by oliversalzburg** [comment]
I can reproduce it with both. It makes no difference. I verified it with the standard one earlier.
